src/Appstream/AppstreamClient.php

```diff
@@ -42,4 +42,16 @@
  * @method \Aws\Result updateStack(array $args = [])
  * @method \GuzzleHttp\Promise\Promise updateStackAsync(array $args = [])
  */
-class AppstreamClient extends AwsClient {}
+class AppstreamClient extends AwsClient {
+
+    public function __construct(array $args)
+    {
+        if (!isset($args['endpoint'])) {
```

Comment:
I think `service` rather than `endpoint` should be injected, e.g., `if (empty($args['service'])) { $args['service'] = 'appstream2'; }`. That should take care of the endpoint and transport scheme resolution.